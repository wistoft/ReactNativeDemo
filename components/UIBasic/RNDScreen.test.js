import React from 'react';
import renderer from 'react-test-renderer';

import { Text } from 'react-native';

import RNDScreen from "./RNDScreen";


test('renders correctly', () => {
	const tree = renderer.create(
			<RNDScreen 
						title="TestTitle"
						onMenuPress={() => {}}
					>
				<Text>TestPageContent</Text>
			</RNDScreen>
		);
	
		expect(tree).toMatchSnapshot();

});


test('fails when chrilden are missing', () => {
	console.error = jest.fn();
	
	const tree = renderer.create(
		<RNDScreen 
					title="TestTitle"
					onMenuPress={() => {}}
				>
		</RNDScreen>
	);
	
	expect(console.error).toHaveBeenCalledTimes(1);
});


test('fails when title is missing', () => {
	console.error = jest.fn();
	
	const tree = renderer.create(
		<RNDScreen 
					onMenuPress={() => {}}
				>
			<Text>TestPageContent</Text>
		</RNDScreen>
	);
	
	expect(console.error).toHaveBeenCalledTimes(1);
});


test('fails when onMenuPress is missing', () => {
	console.error = jest.fn();
	
	const tree = renderer.create(
		<RNDScreen 
					title="TestTitle"
				>
			<Text>TestPageContent</Text>
		</RNDScreen>
	);
	
	//called twice, because Button component also
	// logs a warning.
	expect(console.error).toHaveBeenCalledTimes(2);
});
