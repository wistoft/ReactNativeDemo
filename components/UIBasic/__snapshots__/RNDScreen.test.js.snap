// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "marginTop": 24,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#eee",
        "borderBottomWidth": 1,
        "borderColor": "#ddd",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "padding": 5,
        "paddingLeft": 15,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Menu
        </Text>
      </View>
    </View>
    <Text
      style={
        Object {
          "fontSize": 17,
          "paddingLeft": 10,
        }
      }
    >
      TestTitle
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <Text>
      TestPageContent
    </Text>
  </View>
</View>
`;
